core/ts/MaciState.ts

```diff
@@ -575,10 +574,11 @@ class Poll {
             deactivatedKeyEvent.c2,
         );
 
-        this.deactivatedKeyEvents.forEach(dke => {
-            const deactivatedLeafHash = hash5([dke.keyHash, ...dke.c1, ...dke.c2]);
-            this.deactivatedKeysTree.insert(deactivatedLeafHash) 
-        });
+        if (this.deactivatedKeysTree.nextIndex === 0)
```

Comment:
This is introduced because the this.deactivateKeysTree is already populated since you are maintaining the macistate across different commands?

Reply:
Correct, to avoid to populate twice the `deactivatedKeysTree`